### PR TITLE
VC-4213 add processing register user

### PR DESCRIPTION
**Pull request by alex-bodnar**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by PavelVlasenkoGo** [comment]
LGTM

File: modules/vipcoin/chain/accounts/handle_msg.go [hunk 1]
```diff
@@ -15,5 +20,44 @@ func (m *Module) HandleMsg(index int, msg sdk.Msg, tx *juno.Tx) error {
...
+		return types.ErrInvalidPublicKeyField
+	}
+
+	newAcc := types.Account{
+		Hash:      msg.Hash,
+		Address:   msg.Address,
+		Kinds:     []types.AccountKind{types.ACCOUNT_KIND_GENERAL},
+		State:     types.ACCOUNT_STATE_ACTIVE,
+		PublicKey: publicKeyAny,
+		Wallets:   []string{msg.HolderWallet},
```
Comment (IamStubborN) [inline modules/vipcoin/chain/accounts/handle_msg.go]: ref_reward_wallet must be here too

File: modules/vipcoin/chain/accounts/handle_msg.go [hunk 2]
```diff
@@ -15,5 +20,44 @@ func (m *Module) HandleMsg(index int, msg sdk.Msg, tx *juno.Tx) error {
 		return nil
 	}
 
+	switch accountMsg := msg.(type) {
+	case *types.MsgRegisterUser:
+		return m.handleMsgRegisterUser(tx, index, accountMsg)
+	default:
+		errMsg := fmt.Sprintf("unrecognized %s message type: %T", types.ModuleName, accountMsg)
+		fmt.Println(sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, errMsg))
+		// return sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, errMsg)
```
Comment (IamStubborN) [inline modules/vipcoin/chain/accounts/handle_msg.go:29]: debug info?

File: _tests/database/accounts/register-user_test.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	extratypes "git.ooo.ua/vipcoin/chain/x/types"
+	"git.ooo.ua/vipcoin/lib/filter"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	accountsdb "github.com/forbole/bdjuno/v2/database/vipcoin/chain/accounts"
+	_ "github.com/jackc/pgx/stdlib"
+	"github.com/jmoiron/sqlx"
+)
+
+func TestRepository_SaveRegisterUser(t *testing.T) {
+	// db, err := sqlx.Connect("pgx", "host=10.10.1.79 port=5432 user=postgres dbname=juno password=postgres sslmode=disable")
```
Comment (IamStubborN) [inline _tests/database/accounts/register-user_test.go]: remove

**Review by IamStubborN — CHANGES_REQUESTED** [review]

File: modules/vipcoin/chain/accounts/handle_msg.go (hunk 2 above)
Comment (alex-bodnar) [reply modules/vipcoin/chain/accounts/handle_msg.go:29]: Будет возвращаться ошибка когда сделаем обработчики для всех сообщений, пока, что бы error-ы не сыпались - закоментировал

**Review by IamStubborN — APPROVED** [review]
Lgtm

**Pull request merged**